webpack.config.js

```diff
@@ -19,6 +19,8 @@ module.exports = (env, argv) => {
         // This makes the sourcemaps human readable for developers. We use eval-source-map
         // because the plain source-map devtool ruins the alignment.
         development['devtool'] = 'eval-source-map';
+    } else {
+        development['devtool'] = 'nosources-source-map';
```

Comment:
Why not just `source-map` here? Then browser tools will also show the original source in production as well as dev. Is the reason for `nosources` to reduce file size, or something else...?

Reply:
`source-map` has a history of hostile RAM usage, though that might be a macOS problem only.

Reply:
Once upon a time, that did affect me on macOS, but a quick test suggests it's since been fixed in some part of the stack now.

Reply:
Not too long ago we re-enabled its usage in the pipeline and it took ~3 days for it to consume someone's machine :(

I don't have confidence in it, even if it does work.

Reply:
I suppose it just seems unnecessary? the purpose here is mainly to make stacktraces readable again, and this would increase the bundle size.

Reply:
Well, it doesn't increase the bundle size for a regular user, since the source maps are separate files in the modes we are discussing. It only means larger size for tools and developers who download the separate map file.

I guess I am hearing people aren't that interested in debugging on production then...? That's a bit surprising to me, but I suppose it should be a separate change from whatever's needed for the decoder ring tool.

Reply:
That's a fair point. There's no real reason not to change it?

Reply:
err... very interested in debugging production, just not in production. The sourcemaps likely already end up in the bundle as we have a careful amount of webpack to make sure they aren't separate files.

Reply:
> very interested in debugging production, just not in production.

Assuming the second "in" is "on"... why not? Isn't it nice to be able to open a debugger on /app if needed and see original sources?

> The sourcemaps likely already end up in the bundle as we have a careful amount of webpack to make sure they aren't separate files.

Hmm, I am not aware of anything that would cause that, are you sure...? A quick test of running `yarn build` with `source-map` shows things like:

```
-rw-r--r--  1 jryans  staff    15K Apr 28 16:46 bundle.js
-rw-r--r--  1 jryans  staff   349B Apr 28 16:46 bundle.js.LICENSE.txt
-rw-r--r--  1 jryans  staff    53K Apr 28 16:46 bundle.js.map
-rw-r--r--  1 jryans  staff   524B Apr 28 16:46 error-view.js
-rw-r--r--  1 jryans  staff   1.8K Apr 28 16:46 error-view.js.map
-rw-r--r--  1 jryans  staff    97K Apr 28 16:46 indexeddb-worker.js
-rw-r--r--  1 jryans  staff   120K Apr 28 16:46 indexeddb-worker.js.map
-rw-r--r--  1 jryans  staff   1.9M Apr 28 16:46 init.js
-rw-r--r--  1 jryans  staff   7.5M Apr 28 16:46 init.js.map
```

so the larger map files seem to remain separate.

Reply:
> Assuming the second "in" is "on"... why not? Isn't it nice to be able to open a debugger on /app if needed and see original sources?

in vs on doesn't really make a difference here - the idea is that we have reproducible builds (or better yet, code that is small enough where we don't care that we're including source maps or don't even need them) so we can pull the information offline. It's more a matter of principle: debugging tools, in my opinion, should be available for production however they should not be always deployed alongside it. 

This is quickly turning into something that's hard to convey over github comments though :(

> Hmm, I am not aware of anything that would cause that, are you sure...?

Fairly confident we have that, though I can believe we duplicate the files needlessly. We never really got a chance to finish the webpack work which makes this all work correctly.

Reply:
We definitely want the maps in the bundle –– that way, we can figure out rageshakes by looking at the sourcemaps of the build the user's working against rather than something "nearby"

Reply:
Seems like we may need to discuss in a more real-time medium on to resolve lingering differences here... For the purpose of the decoder ring tool, it's okay as shown, since that's the minimum needed for the tool.